### PR TITLE
Add `/refresh` Endpoint for Refreshing Tokens

### PR DESCRIPTION
**Pull request by samgomena**
This PR adds functionality for:
*  A new endpoint `auth/refresh` which can be used for exchanging a valid token for another valid token with a current `iat` datetime.
* JWT payloads now include an `iss` and `exp` field for specifying token issuer and expiry, respectively.
* Other minor refactorings (e.g. changing `let` variables to `const`.

### CODE REVIEW

File: api/controllers/auth_controller.js [hunk 1]
```diff
@@ -2,9 +2,15 @@
 
 const config = require('../config/config');
 
-let jwt = require('jsonwebtoken');
-let bcrypt = require('bcrypt');
-let customer = require('../models/account_model');
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+const customer = require('../models/account_model');
```
Comment (logyball) [inline api/controllers/auth_controller.js:7]: nice

File: docker-compose.yml [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+#    image: nginx:1.15-alpine
+#    ports:
+#      - "80:80"
+#      - "443:443"
+#    volumes:
+#      - ./conf/nginx:/etc/nginx/conf.d
+
   nginx:
-    image: nginx:1.15-alpine
+    image: owasp/modsecurity
```
Comment (logyball) [inline docker-compose.yml:13]: why swap this?

File: api/controllers/auth_controller.js [hunk 3]
```diff
@@ -2,9 +2,15 @@
...
+
+const jwt_options = {
+    expiresIn: "30d",
```
Comment (logyball) [inline api/controllers/auth_controller.js]: 30 days is pretty long... for a desktop application?  I think that we should maybe use this for testing mostly and not as a default feature

**Review by logyball — APPROVED** [review]
this looks good.  I question the need to issue a refresh token valid for 30 days every time the api is invoked from PLUTO, with each invokation issue a new token, but I'm probably just being over-cautious

File: api/controllers/auth_controller.js (hunk 3 above)
Comment (samgomena) [reply api/controllers/auth_controller.js]: Ahh yea, I'm not really sure what a good length would be. Do you have a suggestion for this value?

File: docker-compose.yml (hunk 2 above)
Comment (samgomena) [reply docker-compose.yml:13]: Ah shit, I didn't mean to include this (again) but I guess it's moot as we're not using the nginx container anyways. It was changed to use a `modsecurity` container that would sit "on top" of an nginx container. I think I committed this while testing it out on accident.

File: api/controllers/auth_controller.js (hunk 3 above)
Comment (logyball) [reply api/controllers/auth_controller.js]: no idea lol maybe 1 hour?

Comment (samgomena) [reply api/controllers/auth_controller.js]: I'm gonna make it a day just for ease of development, we can shorten it or add a "remember me for x days" or something like that down the road.

**Pull request merged**